From 41af0e8a2447242bdbaedbfc119af83842619a61 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Mon, 17 Mar 2025 15:13:19 -0400
Subject: [PATCH 06/38] open.c: add igloo_open_hc

---
 fs/open.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/fs/open.c b/fs/open.c
index 9921f70bc5..309a355ef8 100644
--- a/fs/open.c
+++ b/fs/open.c
@@ -1035,6 +1035,11 @@ struct file *filp_clone_open(struct file *oldfile)
 }
 EXPORT_SYMBOL(filp_clone_open);
 
+#ifdef CONFIG_IGLOO
+// forward declare for igloo_hc_open
+void igloo_hc_open(int dfd, struct filename *tmp, int fd);
+#endif
+
 long do_sys_open(int dfd, const char __user *filename, int flags, umode_t mode)
 {
 	struct open_flags op;
@@ -1048,6 +1053,10 @@ long do_sys_open(int dfd, const char __user *filename, int flags, umode_t mode)
 	if (IS_ERR(tmp))
 		return PTR_ERR(tmp);
 
+#ifdef CONFIG_IGLOO
+	igloo_hc_open(dfd, tmp, fd);
+#endif
+
 	fd = get_unused_fd_flags(flags);
 	if (fd >= 0) {
 		struct file *f = do_filp_open(dfd, tmp, &op);
