From 9e0524b6f487ec769fee9532685d9d62af677248 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Sat, 20 Sep 2025 18:08:14 -0400
Subject: [PATCH 22/38] igloo_weaksyms: fix igloo_hc_open type

---
 drivers/igloobase/igloo_weaksyms.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/igloobase/igloo_weaksyms.c b/drivers/igloobase/igloo_weaksyms.c
index 8498d15500..9e7e5c6639 100644
--- a/drivers/igloobase/igloo_weaksyms.c
+++ b/drivers/igloobase/igloo_weaksyms.c
@@ -63,12 +63,11 @@ void igloo_hc_newuname(struct new_utsname *name)
 }
 EXPORT_SYMBOL(igloo_hc_newuname);
 
-void (*igloo_hc_open_module)(int dfd, const char __user *filename, struct open_how *how);
-void igloo_hc_open(int dfd, const char __user *filename, struct open_how *how);
-void igloo_hc_open(int dfd, const char __user *filename, struct open_how *how)
+void (*igloo_hc_open_module)(int dfd, struct filename *tmp, int fd);
+void igloo_hc_open(int dfd, struct filename *tmp, int fd)
 {
     if (igloo_hc_open_module) {
-        igloo_hc_open_module(dfd, filename, how);
+        igloo_hc_open_module(dfd, tmp, fd);
     } else {
         // printk(KERN_EMERG "igloo_hc_open: unimplemented\n");
     }
