From 757a088b0305f5aa512b02150ed01a60206ccf89 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Sat, 20 Sep 2025 18:09:48 -0400
Subject: [PATCH 27/38] socket.c: add config_igloo

---
 net/socket.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/net/socket.c b/net/socket.c
index 75aa88e267..47ffdfc03a 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -1392,8 +1392,10 @@ SYSCALL_DEFINE4(socketpair, int, family, int, type, int, protocol,
 	return err;
 }
 
+#ifdef CONFIG_IGLOO
 // forward declare igloo_sock_bind
 void igloo_sock_bind(struct socket *sock, struct sockaddr_storage *address);
+#endif
 
 /*
  *	Bind a name to a socket. Nothing much to do here since it's
@@ -1420,7 +1422,9 @@ SYSCALL_DEFINE3(bind, int, fd, struct sockaddr __user *, umyaddr, int, addrlen)
 				err = sock->ops->bind(sock,
 						      (struct sockaddr *)
 						      &address, addrlen);
-				igloo_sock_bind(sock, address);
+				#ifdef CONFIG_IGLOO
+				igloo_sock_bind(sock, &address);
+				#endif
 			}
 		}
 		fput_light(sock->file, fput_needed);
