From 2a7c6f4c78a884f385d23e21f01f3330d796e131 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Mon, 22 Sep 2025 10:57:59 -0400
Subject: [PATCH 30/38] syscalls_info_report: handle different 4.10 issues

---
 drivers/igloobase/syscalls_info_report.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/igloobase/syscalls_info_report.c b/drivers/igloobase/syscalls_info_report.c
index 1f447bcc95..917a1041e1 100644
--- a/drivers/igloobase/syscalls_info_report.c
+++ b/drivers/igloobase/syscalls_info_report.c
@@ -121,9 +121,16 @@ static void report_syscall(char * buffer, struct syscall_metadata *meta){
 #ifdef CONFIG_COMPAT
 /* For ARM64 */
 #if defined(CONFIG_ARM64)
+#ifndef syscall_fn_t
+typedef void* syscall_fn_t;
+#endif
 extern const syscall_fn_t compat_sys_call_table[];
 /* Don't redeclare sys_call_table as it's already in syscall.h with correct type */
+#ifdef __NR_compat32_syscalls
 #define COMPAT_TABLE_SIZE __NR_compat32_syscalls
+#else
+#define COMPAT_TABLE_SIZE __NR_compat_syscalls
+#endif
 
 /* For x86_64 */
 // #elif defined(CONFIG_X86_64)
