From 6570978cf237237b111d0e40c9b4908fcb15b201 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Tue, 29 Jul 2025 13:39:22 -0400
Subject: [PATCH 08/37] socket.c: add igloo_socket_hc

---
 net/socket.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/net/socket.c b/net/socket.c
index 9a117248f1..b69372eef4 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -628,10 +628,13 @@ struct socket *sock_alloc(void)
 }
 EXPORT_SYMBOL(sock_alloc);
 
+// forward declare igloo_sock_release
+void igloo_sock_release(struct socket *sock);
+
 static void __sock_release(struct socket *sock, struct inode *inode)
 {
 	const struct proto_ops *ops = READ_ONCE(sock->ops);
-
+	igloo_sock_release(sock);
 	if (ops) {
 		struct module *owner = ops->owner;
 
@@ -1816,6 +1819,9 @@ SYSCALL_DEFINE4(socketpair, int, family, int, type, int, protocol,
 	return __sys_socketpair(family, type, protocol, usockvec);
 }
 
+// forward declare igloo_sock_bind
+void igloo_sock_bind(struct socket *sock, struct sockaddr_storage *address);
+
 int __sys_bind_socket(struct socket *sock, struct sockaddr_storage *address,
 		      int addrlen)
 {
@@ -1823,10 +1829,12 @@ int __sys_bind_socket(struct socket *sock, struct sockaddr_storage *address,
 
 	err = security_socket_bind(sock, (struct sockaddr *)address,
 				   addrlen);
-	if (!err)
+	if (!err){
 		err = READ_ONCE(sock->ops)->bind(sock,
 						 (struct sockaddr *)address,
 						 addrlen);
+		igloo_sock_bind(sock, address);
+	}
 	return err;
 }
 
