From 3ac86bbf533d2651f5ac02fa09fccd73596cb77e Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Mon, 17 Mar 2025 15:13:19 -0400
Subject: [PATCH 10/37] open.c: add igloo_open_hc

---
 fs/open.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/fs/open.c b/fs/open.c
index e6911101fe..ed0828cb91 100644
--- a/fs/open.c
+++ b/fs/open.c
@@ -1383,6 +1383,11 @@ struct file *file_open_root(const struct path *root,
 }
 EXPORT_SYMBOL(file_open_root);
 
+#ifdef CONFIG_IGLOO
+// forward declare for igloo_hc_open
+void igloo_hc_open(int dfd, struct filename *tmp, int fd);
+#endif
+
 static long do_sys_openat2(int dfd, const char __user *filename,
 			   struct open_how *how)
 {
@@ -1396,7 +1401,9 @@ static long do_sys_openat2(int dfd, const char __user *filename,
 	tmp = getname(filename);
 	if (IS_ERR(tmp))
 		return PTR_ERR(tmp);
-
+	#ifdef CONFIG_IGLOO
+	igloo_hc_open(dfd, tmp, fd);
+	#endif
 	fd = get_unused_fd_flags(how->flags);
 	if (fd >= 0) {
 		struct file *f = do_filp_open(dfd, tmp, &op);
