From eb2e953c1def35bb92deeec5763667ddbc9e31ec Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Mon, 17 Mar 2025 15:12:52 -0400
Subject: [PATCH 13/37] ioctl.c: add igloo_ioctl_hc

---
 fs/ioctl.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/fs/ioctl.c b/fs/ioctl.c
index 638a36be31..270ad13437 100644
--- a/fs/ioctl.c
+++ b/fs/ioctl.c
@@ -789,6 +789,11 @@ static int ioctl_get_fs_sysfs_path(struct file *file, void __user *argp)
 	return copy_to_user(argp, &u, sizeof(u)) ? -EFAULT : 0;
 }
 
+#ifdef CONFIG_IGLOO
+// forward declare igloo_ioctl
+void igloo_ioctl(int error, struct inode *inode, struct file *filp, unsigned int cmd, void __user * argp);
+#endif
+
 /*
  * do_vfs_ioctl() is not for drivers and not intended to be EXPORT_SYMBOL()'d.
  * It's just a simple helper for sys_ioctl and compat_sys_ioctl.
@@ -799,8 +804,8 @@ static int ioctl_get_fs_sysfs_path(struct file *file, void __user *argp)
  * The LSM mailing list should also be notified of any command additions or
  * changes, as specific LSMs may be affected.
  */
-static int do_vfs_ioctl(struct file *filp, unsigned int fd,
-			unsigned int cmd, unsigned long arg)
+static int do_vfs_ioctl(struct file *filp, unsigned int fd, unsigned int cmd,
+			unsigned long arg)
 {
 	void __user *argp = (void __user *)arg;
 	struct inode *inode = file_inode(filp);
@@ -881,8 +886,13 @@ static int do_vfs_ioctl(struct file *filp, unsigned int fd,
 		return ioctl_get_fs_sysfs_path(filp, argp);
 
 	default:
-		if (S_ISREG(inode->i_mode))
-			return file_ioctl(filp, cmd, argp);
+		if (S_ISREG(inode->i_mode)){
+			int error = file_ioctl(filp, cmd, argp);
+			#ifdef CONFIG_IGLOO
+			igloo_ioctl(error, inode, filp, cmd, argp);
+			#endif
+			return error;
+		}
 		break;
 	}
 
