From ca6ce59643e2995add53445a7f05ee8ee117b017 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Fri, 1 Aug 2025 11:15:02 -0400
Subject: [PATCH 15/37] exec.c: add igloo_task_size

---
 fs/exec.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/fs/exec.c b/fs/exec.c
index 98cb7ba998..db46361140 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -72,6 +72,7 @@
 #include <linux/uaccess.h>
 #include <asm/mmu_context.h>
 #include <asm/tlb.h>
+#include <igloo.h>
 
 #include <trace/events/task.h>
 #include "internal.h"
@@ -1432,6 +1433,10 @@ void setup_new_exec(struct linux_binprm * bprm)
 	 * some architectures like powerpc
 	 */
 	me->mm->task_size = TASK_SIZE;
+	#ifdef CONFIG_IGLOO
+	if (igloo_task_size)
+		current->mm->task_size = igloo_task_size;
+	#endif
 	up_write(&me->signal->exec_update_lock);
 	mutex_unlock(&me->signal->cred_guard_mutex);
 }
