From 55f001f78b3a7febc729457685856a4b168ab626 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Fri, 1 Aug 2025 11:16:19 -0400
Subject: [PATCH 17/37] util.c: add igloo_task_size

---
 mm/util.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/mm/util.c b/mm/util.c
index 60aa40f612..f5d5f5dac3 100644
--- a/mm/util.c
+++ b/mm/util.c
@@ -25,6 +25,7 @@
 #include <linux/compat.h>
 
 #include <linux/uaccess.h>
+#include <igloo.h>
 
 #include <kunit/visibility.h>
 
@@ -437,7 +438,13 @@ static unsigned long mmap_base(unsigned long rnd, struct rlimit *rlim_stack)
 	 * task. mmap_base starts directly below the stack and grows
 	 * downwards.
 	 */
-	return PAGE_ALIGN_DOWN(mmap_upper_limit(rlim_stack) - rnd);
+	#ifdef CONFIG_IGLOO
+	if(igloo_task_size) {
+		return PAGE_ALIGN_DOWN(igloo_task_size - rnd);
+	} else {
+		return PAGE_ALIGN_DOWN(mmap_upper_limit(rlim_stack) - rnd);
+	}
+	#endif
 #else
 	unsigned long gap = rlim_stack->rlim_cur;
 	unsigned long pad = stack_guard_gap;
