From eb4971ad09ede9ff4f669fa6a692e58e9c9a05b2 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Fri, 19 Sep 2025 16:57:21 -0400
Subject: [PATCH 25/37] compat.h: add define0

---
 include/linux/compat.h | 48 +++++++++++++++++++++++++++++++++---------
 1 file changed, 38 insertions(+), 10 deletions(-)

diff --git a/include/linux/compat.h b/include/linux/compat.h
index 5121820555..c9df1bacd8 100644
--- a/include/linux/compat.h
+++ b/include/linux/compat.h
@@ -42,11 +42,48 @@
 #define __SC_DELOUSE(t,v) ((__force t)(unsigned long)(v))
 #endif
 
+
+// === Igloo Interception Support ===
+#include <igloo_syscall_macros.h>
+
+#ifdef CONFIG_IGLOO
+extern igloo_syscall_enter_t igloo_syscall_enter_hook;
+extern igloo_syscall_return_t igloo_syscall_return_hook;
+#endif
+
 #ifndef COMPAT_SYSCALL_DEFINE0
 #define COMPAT_SYSCALL_DEFINE0(name) \
 	asmlinkage long compat_sys_##name(void); \
 	ALLOW_ERROR_INJECTION(compat_sys_##name, ERRNO); \
-	asmlinkage long compat_sys_##name(void)
+	asmlinkage long compat_sys_##name(void)	\
+	{										\
+		const char *syscall_basename = __stringify(name); /* Base name */ \
+		long ret;							\
+		bool skip = false;					\
+		long skip_ret = 0;					\
+		/* Array for arguments (empty for 0 args) */ \
+		unsigned long args_array[IGLOO_SYSCALL_MAXARGS] = {0}; \
+		/* === Igloo Enter Hook === */ \
+		if (igloo_syscall_enter_hook) {	\
+			/* Pass NULL for setter func for 0-arg syscalls */ \
+			skip = igloo_syscall_enter_hook(syscall_basename, &skip_ret, 0, args_array, NULL); \
+		}								\
+										\
+		if (skip) {						\
+			ret = skip_ret;				\
+		} else {						\
+			ret = __do_compat_sys_##name(); \
+		}								\
+										\
+		/* === Igloo Return Hook === */ \
+		if (igloo_syscall_return_hook) {	\
+			ret = igloo_syscall_return_hook(syscall_basename, ret, 0, args_array); \
+		}								\
+										\
+		return ret ;					\
+	}	\
+	static inline long __do_compat_sys_##name(void)
+
 #endif /* COMPAT_SYSCALL_DEFINE0 */
 
 #define COMPAT_SYSCALL_DEFINE1(name, ...) \
@@ -62,15 +99,6 @@
 #define COMPAT_SYSCALL_DEFINE6(name, ...) \
 	COMPAT_SYSCALL_DEFINEx(6, _##name, __VA_ARGS__)
 
-
-// === Igloo Interception Support ===
-#include <igloo_syscall_macros.h>
-
-#ifdef CONFIG_IGLOO
-extern igloo_syscall_enter_t igloo_syscall_enter_hook;
-extern igloo_syscall_return_t igloo_syscall_return_hook;
-#endif
-
 /*
  * The asmlinkage stub is aliased to a function named __se_compat_sys_*() which
  * sign-extends 32-bit ints to longs whenever needed. The actual work is
