From 4799b41002283a8608f919dd775d2945839ff168 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Mon, 22 Sep 2025 10:58:40 -0400
Subject: [PATCH 26/37] igloo_weaksyms: fix up igloo_ioctl type

---
 drivers/igloobase/igloo_weaksyms.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/igloobase/igloo_weaksyms.c b/drivers/igloobase/igloo_weaksyms.c
index 8498d15500..33e83342d2 100644
--- a/drivers/igloobase/igloo_weaksyms.c
+++ b/drivers/igloobase/igloo_weaksyms.c
@@ -75,12 +75,12 @@ void igloo_hc_open(int dfd, const char __user *filename, struct open_how *how)
 }
 EXPORT_SYMBOL(igloo_hc_open);
 
-void (*igloo_ioctl_module)(int error, struct inode *inode, struct file *filp, unsigned int cmd, void __user *argp);
+void (*igloo_ioctl_module)(int error, struct file *filp, unsigned int cmd);
 void igloo_ioctl(int error, struct inode *inode, struct file *filp, unsigned int cmd, void __user *argp);
 void igloo_ioctl(int error, struct inode *inode, struct file *filp, unsigned int cmd, void __user *argp)
 {
     if (igloo_ioctl_module) {
-        igloo_ioctl_module(error, inode, filp, cmd, argp);
+        igloo_ioctl_module(error, filp, cmd);
     } else {
         // printk(KERN_EMERG "igloo_ioctl: unimplemented\n");
     }
